Microsoft.Toolkit.Uwp.UI.Controls.DataGrid.Design/MetadataRegistration.cs

```diff
@@ -22,25 +28,17 @@ public MetadataRegistration() : base()
             // (or Default namespace in Project -> Properties -> Application tab)
             // must be the same as runtime assembly's main namespace (t.Namespace)
             // plus .Design.
-            Type t = typeof(Microsoft.Toolkit.Uwp.UI.Controls.DataGrid);
+#if VS_DESIGNER_PROCESS_ISOLATION
+            Type t = typeof(DataGrid);
             AssemblyName an = t.Assembly.GetName();
             AssemblyFullName = ", " + an.FullName;
             XmlResourceName = t.Namespace + ".Design." + an.Name + ".xml";
+#else
+            var RootNamespace = "Microsoft.Toolkit.Uwp.UI.Controls";
+            var AssemblyName = "Microsoft.Toolkit.Uwp.UI.Controls";
+            AssemblyFullName = $"{RootNamespace}, Version=6.1.0.0, Culture=neutral, PublicKeyToken=null";
+            XmlResourceName = $"{RootNamespace}.Design.{AssemblyName}.xml";
```

Comment:
Same here. Anyway to do this without hardcoding it like this?

This is used for providing the Descriptions and discovery attributes by using the XML docs.

**Is there anyway to make it better or remove dependency on `Controls` assembly's assets?**

Reply:
Had some few ideas on how to do this!! Will discuss it in the next PR.